### PR TITLE
Removing archive from upload for files.

### PR DESCRIPTION
**Pull request by bhargav191098**
- Removing archive step for files during upload: Users can upload files.
- Making changes in the download logic to accommodate this as well.
- For directories - the old behavior of zip is retained.

### CODE REVIEW

File: python/fedml/api/modules/storage.py [hunk 1]
```diff
@@ -38,27 +38,40 @@ def upload(data_path, api_key, name, description, tag_list, service, show_progre
...
-    archive_path, message = _archive_data(data_path)
-    if not archive_path:
+    if(data_type == "dir"):
+        to_upload_path, message = _archive_data(data_path)
+        name = os.path.splitext(os.path.basename(to_upload_path))[0] if name is None else name
+        file_name = name + ".zip"
+    else:
+        to_upload_path = data_path
+        base_name = os.path.basename(to_upload_path)
+        given_extension = os.path.splitext(name)[1]
```
Comment (Raphael-Jin) [inline python/fedml/api/modules/storage.py]: Be mindful when name is None

Comment (Raphael-Jin) [reply python/fedml/api/modules/storage.py]: You might want retrieve the type from the suffix of data_path, e.g. abc.png -> png

**Review by Raphael-Jin — APPROVED** [review]
LGTM

File: python/fedml/api/modules/storage.py [hunk 2]
```diff
@@ -438,10 +469,12 @@ def _get_storage_service(service):
...
-def _check_data_path(data_path):
-    if os.path.isdir(data_path) or os.path.isfile(data_path):
-        return True
-    return False
+def _get_data_type(data_path):
+    if os.path.isdir(data_path):
+        return "dir"
+    elif os.path.isfile(data_path):
+        return "file"
+    return "invalid"
```
Comment (alaydshah) [inline python/fedml/api/modules/storage.py]: Let's use enums here

Comment (bhargav191098) [reply python/fedml/api/modules/storage.py]: I've added and tested. Let me know if that was the expectation.

**Review by alaydshah — APPROVED** [review]

**Pull request merged**